### DISCUSSION TITLE
Website with dynamic data on vercel as static website.

### DISCUSSION THREAD

**Discussion by muke5hy** [vercel/next.js#16472] [Help] [answered]
Hi, I have a requirement where I'll be having 7-8 pages as static and blog which can be easily built with NextJs and I have done the tutorial for the same. 

Along with the static website, there is a dashboard which is user-specific and it contains dynamic data, with user authentication. 

I'm not able to find any tutorial which can cater to these requirements. I want to use strapi as REST API to display the dashboard.

**Comment by grsoares21** [comment, marked as answer]
You can have both static and dynamically rendered pages in the same Next.js project.
For the pages that are static, you have the `getStaticProps` function but you probably already know this as you said that you already have the static generated blog. These will be rendered at build time.

For pages that are dynamic, you can either use the `getServerSideProps` if you want to render this page on the server-side or use a normal `fetch` in your react component code to contact your API from the front-end. 

Note that this decision is on a per-page basis, you can definitely do both on the same project.

**Reply by muke5hy** [reply]
Thank you for your response, I have started the project. 
I have another question which I'll post in another thread. Just for reference, I have got one dashboard https://github.com/coreui/coreui-free-react-admin-template which I want to use for dashboard. Not sure creating Nextjs component from the React component will be easy than creating from a plain bootstrap template. 

Thanks
